pkg/app/api/insightstore/report.go

```diff
@@ -0,0 +1,188 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package insightstore
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/pipe-cd/pipe/pkg/model"
+)
+
+// deploy frequency
+
+// deployFrequencyReport satisfy the interface `Report`
+type deployFrequencyReport struct {
+	AccumulatedTo int64                    `json:"accumulated_to"`
+	Datapoints    deployFrequencyDataPoint `json:"datapoints"`
+	FilePath      string
+}
+
+type deployFrequencyDataPoint struct {
+	Daily   map[string]deployFrequency `json:"daily"`
+	Weekly  map[string]deployFrequency `json:"weekly"`
+	Monthly map[string]deployFrequency `json:"monthly"`
+	Yearly  map[string]deployFrequency `json:"yearly"`
+}
+
+type deployFrequency struct {
+	DeployCount float32 `json:"deploy_count"`
+}
+
+func (d *deployFrequencyReport) GetFilePath() string {
+	return d.FilePath
+}
+
+func (d *deployFrequencyReport) PutFilePath(path string) {
+	d.FilePath = path
+}
+
+func (d *deployFrequencyReport) Value(step model.InsightStep, key string) (float32, error) {
+	switch step {
+	case model.InsightStep_YEARLY:
+		return d.Datapoints.Yearly[key].DeployCount, nil
+	case model.InsightStep_MONTHLY:
+		return d.Datapoints.Monthly[key].DeployCount, nil
+	case model.InsightStep_WEEKLY:
+		return d.Datapoints.Weekly[key].DeployCount, nil
+	case model.InsightStep_DAILY:
+		return d.Datapoints.Daily[key].DeployCount, nil
+	}
+	return 0, fmt.Errorf("value not found. step: %d, key: %s", step, key)
+}
+
+// change failure rate
+
+// changeFailureRateReport satisfy the interface `Report`
+type changeFailureRateReport struct {
+	AccumulatedTo int64                      `json:"accumulated_to"`
+	Datapoints    changeFailureRateDataPoint `json:"datapoints"`
+	FilePath      string
+}
+
+type changeFailureRateDataPoint struct {
+	Daily   map[string]changeFailureRate `json:"daily"`
+	Weekly  map[string]changeFailureRate `json:"weekly"`
+	Monthly map[string]changeFailureRate `json:"monthly"`
+	Yearly  map[string]changeFailureRate `json:"yearly"`
+}
+
+type changeFailureRate struct {
+	Rate         float32 `json:"rate"`
+	SuccessCount int64   `json:"success_count"`
+	FailureCount int64   `json:"failure_count"`
+}
+
+func (c *changeFailureRateReport) GetFilePath() string {
+	return c.FilePath
+}
+
+func (c *changeFailureRateReport) PutFilePath(path string) {
+	c.FilePath = path
+}
+
+func (c *changeFailureRateReport) Value(step model.InsightStep, key string) (float32, error) {
+	switch step {
+	case model.InsightStep_YEARLY:
+		return c.Datapoints.Yearly[key].Rate, nil
+	case model.InsightStep_MONTHLY:
+		return c.Datapoints.Monthly[key].Rate, nil
+	case model.InsightStep_WEEKLY:
+		return c.Datapoints.Weekly[key].Rate, nil
+	case model.InsightStep_DAILY:
+		return c.Datapoints.Daily[key].Rate, nil
+	}
+	return 0, fmt.Errorf("value not found. step: %d, key: %s", step, key)
+}
+
+type Report interface {
+	// GetFilePath get filepath
```

Comment:
All interface comments are preferable to be in the third person singular.
```suggestion
	// GetFilePath gets filepath
```